rgo/component.py

```diff
@@ -135,40 +139,35 @@ def _make_srpm(tmpdir, distgit, source_code_git, pkg_name, version_from, patches
                                check=True, input=proc.stdout, stdout=f_archive)
 
             # Prepare new spec
-            with open(_spec_path, "w") as specfile:
+            with open(_final_spec_path, "w") as specfile:
                 prepared = "{!s}\n{!s}".format(
-                    utils.prepare_spec(spec, version, release, nvr, patches_action),
-                    utils.generate_changelog(source_code_git.timestamp, version, release, source_code_git.get_rpm_changelog()))
+                    utils.prepare_spec(original_spec_path, version, release, nvr, patches_action),
+                    utils.generate_changelog(source_code_git.timestamp, version, release,
+                                             source_code_git.get_rpm_changelog()))
                 specfile.write(prepared)
         else:
             # Just copy spec from distgit
-            shutil.copy2(spec, _spec_path)
-        spec = _spec_path
-
-        _sources = []
-        for src, _, src_type in rpm.spec(spec).sources:
-            if src_type == rpm.RPMBUILD_ISPATCH:
-                if patches_action == PatchesAction.keep:
+            shutil.copy2(original_spec_path, _final_spec_path)
+            raise NotImplementedError("Getting a tarball from distgit is not implemented")
```

Comment:
Is this a mistake? You're copying the spec and then raising an exception directly after that.

Reply:
I have commented it out with some description.